iml-manager-cli/src/manager_cli_error.rs

```diff
@@ -2,14 +2,45 @@
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
 
-pub type Result<T> = std::result::Result<T, ImlManagerCliError>;
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum DurationParseError {
+    NoUnit,
+    InvalidUnit,
+    InvalidValue,
+}
+
+impl std::fmt::Display for DurationParseError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            DurationParseError::NoUnit => write!(f, "No unit specified."),
+            DurationParseError::InvalidUnit => {
+                write!(f, "Invalid unit. Valid units include 'h' and 'd'.")
+            }
+            DurationParseError::InvalidValue => {
+                write!(f, "Invalid value specified. Must be a valid integer.")
+            }
+        }
+    }
+}
+
+impl std::error::Error for DurationParseError {
+    fn description(&self) -> &str {
```

Comment:
`description` is soft-deprecated.

Instead of `description`, you should only need to implement `source`.

Reply:
and actually, since the default for `source` is `None`, you don't even need to implement that. In other words:

```rust
impl std::error::Error for DurationParseError {}
```

should be all you need.

Reply:
even better :)